src/topology/vector_bundle.lean

```diff
@@ -0,0 +1,221 @@
+/-
+Copyright © 2020 Nicolò Cavalleri. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Author: Nicolò Cavalleri.
+-/
+
+import topology.topological_fiber_bundle
+import analysis.calculus.deriv
+import linear_algebra.dual
+
+/-!
+# Topological vector bundles
+
+In this file we define topological vector bundles.
+
+Let `B` be the base space. In our formalism, a topological vector bundle is by definition the type
+`bundle.total_space E` where `E : B → Type*` is a function associating to
+`x : B` the fiber over `x`. This type `bundle.total_space E` is just a type synonym for
+`Σ (x : B), E x`, with the interest that one can put another topology than on `Σ (x : B), E x`
+which has the disjoint union topology.
+
+To have a topological vector bundle structure on `bundle.total_space E`,
+one should addtionally have the following data:
+
+* `F` should be a topological vector space over a field `𝕜`;
+* There should be a topology on `bundle.total_space E`, for which the projection to `E` is
+a topological fiber bundle with fiber `F` (in particular, each fiber `E x` is homeomorphic to `F`);
+* For each `x`, the fiber `E x` should be a topological vector space over `𝕜`, and the injection
+from `E x` to `bundle.total_space F E` should be an embedding;
+* The most important condition: around each point, there should be a bundle trivialization which
+is a continuous linear equiv in the fibers.
+
+If all these conditions are satisfied, we register the typeclass
+`topological_vector_bundle 𝕜 F E`. We emphasize that the data is provided by other classes, and
+that the `topological_vector_bundle` class is `Prop`-valued.
+
+The point of this formalism is that it is unbundled in the sense that the total space of the bundle
+is a type with a topology, with which one can work or put further structure, and still one can
+perform operations on topological vector bundles (which are yet to be formalized). For instance,
```

Comment:
What is "yet to be formalized"? Is this referring to the operations rather than the bundles?

Reply:
I am not sure I wrote this but I think it does refer to the operations. Does it need to be changed? I guess it is clear from the context that grammatically it refers to the whole "operations on topological vector bundles"

Reply:
Sure.